src/student_success_tool/modeling/bias_detection.py

```diff
@@ -0,0 +1,213 @@
+import pandas as pd
+import numpy as np
+import scipy.stats as st
+from sklearn.metrics import confusion_matrix
+
+# FNPR sample threshold
+MIN_FNPR_SAMPLES = 50
+Z = st.norm.ppf(1 - (1 - 0.95) / 2)
+
+# Flag FNPR difference thresholds
+HIGH_FLAG_THRESHOLD = 0.15
+MODERATE_FLAG_THRESHOLD = 0.1
+LOW_FLAG_THRESHOLD = 0.05
+
+# Define flag types
+FLAG_NAMES = {
+    "🟢 NO BIAS": "no_bias",
+    "🟡 LOW BIAS": "low_bias",
+    "🟠 MODERATE BIAS": "moderate_bias",
+    "🔴 HIGH BIAS": "high_bias",
+}
+
+
+def calculate_fnpr_and_ci(
+    targets: pd.Series,
+    preds: pd.Series,
```

Comment:
This is potentially confusing. I'm assuming `targets` represents the "true" / known outcomes, while `preds` is the model predictions? Are those going to be binary True/False or probabilities? Describing in the docstring would be a big help.

Reply:
I stand by my comment -- that reads to me as "potentially confusing" :)